thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2911,10 +3036,21 @@ class HashDedupSlaveActivityBase : public CSlaveActivity
         ActivityTimer t(totalCycles, timeActivities);
         if (eos)
             return NULL;
+        if (keepBestRowsReadyToReturn)
+        {
+            const void * row = bestRowStream->nextRow();
+            if (row)
+            {
+                LinkThorRow(row);
+                return row;
+            }
+            // drop-through
```

Comment:
I am not sure of the logic here.
When exhausted bestRowStream, looks like it will fall through to getNextBucketHanlder loop, but the bucket handler it was working on will now have been read/flushed.

This is related to the other comments re. spilling.